runtime/mrc/src/lib.rs

```diff
@@ -693,6 +694,16 @@ impl_runtime_apis! {
 		fn get_msa_id(key: AccountId) -> Result<Option<MessageSenderId>, DispatchError> {
 			Ok(Msa::get_owner_of(&key))
 		}
+
+		fn check_delegations(delegator_msa_ids: Vec<MessageSenderId>, provider_msa_id: MessageSenderId) -> Result<BTreeMap<MessageSenderId, bool>, DispatchError> {
+			let mut map = BTreeMap::new();
+			let provider = pallet_msa::types::Provider(provider_msa_id);
+			for id in delegator_msa_ids {
+				let delegator = pallet_msa::types::Delegator(id);
```

Comment:
@dopatraman  you can use rayon to make it a parallel iterator

Reply:
And move this to rpc call itself if doable .
An example https://github.com/LibertyDSNP/mrc/blob/5b2a52db5ee681e9f36c630b7ffa2fce4c7aec28/common/helpers/src/avro.rs#L58

But if it's not doable due to rayon it's fine as it is 

Just a suggestion